syntax = "proto3";

package user;

import "google/protobuf/empty.proto";

message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
}

message CreateUserRequest {
  string name = 1;
  string email = 2;
}

message GetUserRequest {
  int32 id = 1;
}

message ListUsersResponse {
  repeated User users = 1;
}

service UserService {
  rpc CreateUser (CreateUserRequest) returns (User);
  rpc GetUser (GetUserRequest) returns (User);
  rpc ListUsers (google.protobuf.Empty) returns (ListUsersResponse);
}
